### PR TITLE
Try to add a nfft test

### PR DESCRIPTION
**Pull request by phara92**
- [ ] Tests added / passed

  ```bash
  $ py.test -v -s africanus
  ```

  If the pycodestyle tests fail, the quickest way to correct
  this is to run `autopep8` and then `pycodestyle` to fix the
  remaining issues.

  ```
  $ pip install -U autopep8 pycodestyle
  $ autopep8 -r -i africanus
  $ pycodestyle africanus
  ```

- [ ] Fully documented, including `HISTORY.rst` for all changes
      and one of the `docs/*-api.rst` files for new API

  To build the docs locally:

  ```
  pip install -r requirements.readthedocs.txt
  cd docs
  READTHEDOCS=True make html
  ```

### CODE REVIEW

File: africanus/dft/tests/test_dft.py [hunk 1]
```diff
@@ -8,6 +8,52 @@
 import pytest
 
 
+def test_dft_vs_fftpack():
+    from pynfft.nfft import NFFT
```
Comment (sjperkins) [inline africanus/dft/tests/test_dft.py]:
```suggestion
    NFFT = pytest.importorskip('pynfft.nfft')
```

File: africanus/dft/tests/test_dft.py [hunk 2]
```diff
@@ -8,6 +8,52 @@
...
+    npix = 9
+    nrow = 2
+    uvw = np.random.random(size=(nrow, 3)) - 0.5
+    uvw.T[2] = 0
+    uv = uvw[:, 0:2]
+
+    sky = np.zeros((npix, npix, 1))
+    sky[npix//2+1, npix//2] = 1
+
+    import matplotlib.pyplot as plt
```
Comment (sjperkins) [inline africanus/dft/tests/test_dft.py]: Parametrize the test

```python
@pytest.mark.parametrize('plot', [False])
test_dft_vs_fftpack(plot):
```

Guard the plotting with an try catch block

```python
try:
   import matplotlib.pyplot as plt
except ImportError:
   if plot:
       pytest.fail("Plotting requested but matplotlib not installed")
else:
   # rest of the plotting code
   ...
```

File: africanus/dft/tests/test_dft.py [hunk 3]
```diff
@@ -8,6 +8,52 @@
 import pytest
 
 
+def test_dft_vs_fftpack():
```
Comment (sjperkins) [inline africanus/dft/tests/test_dft.py]: Mark as expected failure

```python
@pytest.mark.xfail(reason="test not yet complete")
def test_dft_vs_fft_pack():
```

**Comment by landmanbester** [comment]
This one has gone dead. I'll add a comparison to the nifty gridder instead

**Pull request closed without merging**